indico/modules/events/contributions/controllers.py

```diff
@@ -113,3 +111,22 @@ def _process(self):
         flash(ngettext("Contribution was deleted.",
                        "{count} contributions were deleted.", deleted_count).format(count=deleted_count), 'success')
         return jsonify_data(html=_render_contribution_list(self.event_new))
+
+
+class RHContributionREST(RHManageContributionBase):
+    def _process_DELETE(self):
+        delete_contribution(self.contrib)
+        flash(_("Contribution '{}' successfully deleted").format(self.contrib), 'success')
+        return jsonify_data(html=_render_contribution_list(self.event_new))
+
+    def _process_PATCH(self):
+        data = request.json
+        updates = {}
+        if data.viewkeys() > 'session_id':
+            raise BadRequest
+        if 'session_id' in data:
+            session = self.event_new.sessions.filter_by(id=data['session_id'], is_deleted=False).first_or_404()
```

Comment:
you want 400, not 404 in case the session doesn't exist. 404 is for wrong stuff in the URL